en/guide/taking_off_landing.md

```diff
@@ -0,0 +1,220 @@
+const Action::Result takeoff_result = device.action().takeoff();
+if (takeoff_result != Action::Result::SUCCESS) {
+    std::cout << "Takeoff failed:" << Action::result_str(
+        takeoff_result) << std::endl;
+    return 1;
+}
+```
+
+If the command succeeds the vehicle will takeoff, and hover at the takeoff altitude. Code should wait until takeoff has completed before sending the next instruction. Unfortunately there is no specific indicator to inform code that takeoff is complete. 
+
+> **Note** One alternative is to simply wait for enough time that the vehicle *should* have reached the takeoff altitude.
+
+The code below checks for takeoff completion by polling the current altitude until the target altitude is reached:
+
+```cpp
+float target_alt=device.action().get_takeoff_altitude_m();
+float current_position=0;
+while (current_position<target_alt) {
+    current_position = device.telemetry().position().relative_altitude_m;
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+}
+// Reached target altitude - continue with next instruction.
+```
+
+
+## Landing
+
+The best way to land the vehicle at the current location is to use the [land()](../api_reference/classdronecore_1_1_action.md#classdronecore_1_1_action_1ad1a50dd7bff99d3099916576efbf8cf6) or [land_async()](../api_reference/classdronecore_1_1_action.md#classdronecore_1_1_action_1a7f10240cde2ff237795e3688802d857b) methods. If the command is accepted the vehicle will change to the [Land mode](https://docs.px4.io/en/flight_modes/land.html) and land at the current point. 
+
+> **Note** DroneCore does not at time of writing recommend other approaches for landing: land mission items are not supported and manually landing the vehicle using the offboard is not as safe. 
+
+The code below shows how to use the land action.
+
+```cpp
+const Action::Result land_result = device.action().land();
+if (land_result != Action::Result::SUCCESS) {
+    //Land failed, so exit (in reality might try a return to land or kill.)
+    return 1;
+}
+```
+
+The vehicle should land and then automatically disarm. If you want to monitor the landing you can trigger completion of the
+app based on the armed state, as shown below.
+
+```cpp
+while (device.telemetry().armed()) {
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+}
+std::cout << "Disarmed, exiting." << std::endl;
+```
+
+
+## Return/RTL
+
+[Return mode](https://docs.px4.io/en/flight_modes/rtl.html) (also known as "Return to Launch", "Return to Land", "Return to Home") flies the vehicle back to the home position and may also land the vehicle (depending on vehicle configuration). This mode is invoked from DroneCore using the [return_to_launch()](../api_reference/classdronecore_1_1_action.md#classdronecore_1_1_action_1a9af73101ce850e37cf7259dcdeda2eb9) or [return_to_launch_async()](../api_reference/classdronecore_1_1_action.md#classdronecore_1_1_action_1aa1253c356c7628d329dfa98d78eb39ee) methods. 

```

Comment:
There is a big discussion on return modes going on now. For this purpose lets just ignore that :-)

Reply:
Yes. There are a couple more possible modes and variations on modes in the works - rally points (return to a safe point) and follow original path back to safety.